File: cp/sorts/insertion.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i-1
        key = arr[i]
        while j >= 0 and arr[j] > key:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key


arr = [3, 21, 6, 35, 7, 0]
print(arr)
insertion_sort(arr)
print(arr)

File: algoritmos e estruturas de dados/001 - Sorts/insertion.c
#include <stdio.h>
#include <stdlib.h>
#define tam 100

int *allocAndGenerate();
void printArray(int *vet, int size,  char sep);

int main(int argc, char const *argv[])
{
    int *arr;

    arr = allocAndGenerate();

    printArray(arr, tam, ' ');

    return 0;
}

int *allocAndGenerate()
{
    int* aloc = (int *)malloc(sizeof(int) * tam);

    for (int i = 0; i < tam; i++)
    {
        aloc[i] = rand() % tam;
    }

    return aloc;
}

void printArray(int *vet, int size, char sep){
    for(int i = 0; i < size; i++){
        printf("%d%c", vet[i], sep);
    }
    printf("\n");
}

void insertionSort(int *vet, int size){
    
}